MediaProcessor/src/FFmpegControllerBuilder.cpp

```diff
@@ -0,0 +1,47 @@
+#include "FFmpegControllerBuilder.h"
+
+#include "FFmpegController.h"
+#include "FFmpegSettingsManager.h"
+
+namespace MediaProcessor {
+
+FFmpegControllerBuilder::FFmpegControllerBuilder() : m_ffmpegSettings(FFmpegSettingsManager()) {}
+
+FFmpegControllerBuilder::FFmpegControllerBuilder(FFmpegSettingsManager& ffmpegSettings)
+    : m_ffmpegSettings(std::move(ffmpegSettings)) {}
+
+FFmpegControllerBuilder& FFmpegControllerBuilder::setOverwrite(bool overwrite) {
+    m_ffmpegSettings.setOverwrite(overwrite);
+
+    return *this;
+}
+
+FFmpegControllerBuilder& FFmpegControllerBuilder::setAudioCodec(AudioCodec codec) {
+    m_ffmpegSettings.setAudioCodec(codec);
+
+    return *this;
+}
+
+FFmpegControllerBuilder& FFmpegControllerBuilder::setAudioSampleRate(int sampleRate) {
+    m_ffmpegSettings.setAudioSampleRate(sampleRate);
+
+    return *this;
+}
+
+FFmpegControllerBuilder& FFmpegControllerBuilder::setAudioChannels(int numChannels) {
+    m_ffmpegSettings.setAudioChannels(numChannels);
+
+    return *this;
+}
+
+FFmpegControllerBuilder& FFmpegControllerBuilder::setVideoCodec(VideoCodec codec) {
+    m_ffmpegSettings.setVideoCodec(codec);
+
+    return *this;
+}
```

Comment:
Sorry I think I'm misunderstanding. I definitely agree about the redundancy, but what do you mean that it handles all the settings? At the moment I'm imagining it would fill a new SettingsManager with the user's settings and return that with the controller, but that also feels redundant.